teps/0044-decouple-task-composition-from-scheduling.md

```diff
@@ -130,6 +132,786 @@ Where they differ:
       address the above use case; for example in PipelineResources, you can have a
       storage "output" but if the steps fail, the "output" pipelineresource will not run
 
+## Design details
+
+TBD - currently focusing on enumerating and examining alternatives before selecting one or more ways forward.
+
+## Alternatives
```

Comment:
Great to see many alternatives!

But if this part now is "shared" with TEP-0046(?) then I think it should be pointed out that only a few of the alternatives solves the problem statement in TEP-0046 that is about Task parallelism, primarily.

Also a `TaskRunGroup` CRD might be a viable addition for some of the alternatives - since some are written from the pipeline authoring view, and some from the runtime view. Note this is different from a TaskGroup CRD that is targeting authors.

Reply:
> then I think it should be pointed out that only a few of the alternatives solves the problem statement in TEP-0046 that is about Task parallelism, primarily

I think this is why it probably still makes sense to have 2 TEPs - tho i think in the long run even the solution for this TEP will involve supporting Tasks in parallel on the same pod

> Also a TaskRunGroup (or TaskGroupRun?) CRD might be a viable addition for some of the alternatives - since some are written from the pipeline authoring view, and some from the runtime view. Note this is different from a TaskGroup CRD that is targeting authors.

I've tried to add some details about this under "Create a new Grouping CRD" but I'm not totally clear on what this would look like - would it basically contain an entire pipeline definition? or link to a pipeline ref?

Reply:
With `TaskGroupRun` CRD  here, I didn't mean a new CRD for Pipeline- or Task authors, more a run-CRD in case the solution should be open for using more than one Pod (here I am assuming an alternative to TaskRun that I assume is strictly for one-to-one relation between a Task and a Pod, whereas TaskGroupRun is a thought about a many-to-one relation between Tasks and a Pod - and there could be one or multiple of TaskGroupRun related to PipelineRun).

Just thinking out loud here - no deeper thoughts.